terraform-plans/modules/GitHub/workflows/main.tf

```diff
@@ -9,36 +9,75 @@ terraform {
 
 provider "github" {
   owner = var.owner
-  app_auth {} # using environmet variables for auth
+  app_auth {} # using environment variables for auth
 }
 
-
 resource "random_string" "update_uid" {
   length  = 8
   numeric = true
   special = false
 }
 
+locals {
+  repo_files = flatten([
+    for file_key, file_info in var.workflow_files : {
+      file = file_info.destination
+      source = file_info.source
+    }
+  ])
+}
+
+data "github_repository_file" "files" {
+  for_each = {
+    for item in local.repo_files : "${item.file}" => item
+  }
+
+  repository = var.repository
+  file       = each.value.file
+  branch     = var.branch
+}
+
+locals {
+  repository_files_content = {
+    for file_key, file_info in var.workflow_files : file_info.destination =>
+      try(data.github_repository_file.files["${file_info.destination}"].content, "")
+  }
+}
+
+locals {
+  changed_files = {
+    for file_key, file_info in var.workflow_files : file_key => file_info
+      if file(file_info.source) != local.repository_files_content[file_info.destination]
+  }
+}
+
 resource "github_branch" "workflows_branch" {
+  count = length(local.changed_files) > 0 ? 1 : 0
+
   repository    = var.repository
   branch        = "${var.pr_branch}-${random_string.update_uid.id}"
   source_branch = var.branch
 }
 
 resource "github_repository_file" "workflows_files" {
-  for_each            = var.workflow_files
+  for_each = local.changed_files
+
   repository          = var.repository
-  branch              = github_branch.workflows_branch.branch
+  branch              = github_branch.workflows_branch[0].branch
```

Comment:
why do we need `[0]`,  `github_branch.workflows_branch` does not look like a list though

Reply:
Because we use `count` in `workflows_branch`